src/StoreKeeper/WooCommerce/B2C/Core.php

```diff
@@ -463,4 +467,99 @@ public function renderBootError(BootError $e)
             }
         }
     }
+
+    /**
+     * @param $settings
+     * @return mixed
+     */
+    public function tsAddExtraFieldsInShippingMethods($settings): mixed {
+        // Add a new setting for Minimum Amount
+        $new_settings = $settings;
+        $currency_code = get_woocommerce_currency_symbol();
+
+        $new_settings['min_amount'] = array(
+            'title'       => __('Minimum Cost (' . $currency_code .')', 'woocommerce'),
+            'type'        => 'text',
+            'class'       => 'currency-input2',
+            'placeholder' => '100.00', // Adjusted to reflect decimal format
+            'default'     => '100.00', // Default value as a decimal
+            'custom_attributes' => array(
+                'min' => 0,
+                'step' => '0.01' // Allow decimal steps in the input
+            ),
+        );
+        return $new_settings;
+    }
+
+    /**
+     * @return void
+     */
+    public function tsDisplayShippingMinAmountContent(): void  {
+        // Retrieve the shipping method selected by the user
+        $chosen_shipping_method = WC()->session->get('chosen_shipping_methods')[0];
+
+        // Get the shipping settings for the chosen method
+        $shipping_method_settings = get_option("woocommerce_{$chosen_shipping_method}_settings");
+
+        // Ensure the settings are an array and contain 'min_amount'
+        if (is_array($shipping_method_settings) && isset($shipping_method_settings['min_amount'])) {
+            $min_amount = floatval($shipping_method_settings['min_amount']);
+        } else {
+            $min_amount = ''; // Fallback if the option or 'min_amount' key is not available
+        }
+    }
+
+
+    /**
+     * @return void
+     */
+    public function tsApplyMinAmountToAllShippingMethods(): void {
+        // Manually add the filter for known methods, including Local Pickup
+        $shipping_methods = array('flat_rate', 'free_shipping', 'local_pickup');
+
+        foreach ($shipping_methods as $method_id) {
+            add_filter("woocommerce_shipping_instance_form_fields_{$method_id}", array($this, 'tsAddExtraFieldsInShippingMethods'), 10, 1);
+        }
+    }
+
+    /**
+     * @param $rates
+     * @param $package
+     * @return mixed
+     */
+    public function tsModifyShippingRates($rates, $package): mixed {
+        // Get the formatted cart total using WooCommerce's functions and settings
+        $total = WC()->cart->get_cart_contents_total();
+        $formatted_cart_total = floatval($total); // WooCommerce already formats the price, so we just cast it to float
```

Comment:
$total is already and float why are we mapping it https://wp-kama.com/plugin/woocommerce/function/WC_Cart::get_cart_contents_total ?